### PR TITLE
[feature] RF-82: Integrate scheduleRoutine

### PR DESCRIPTION
**Pull request by AgustinCarthery**
# [ RF-82: Integrate scheduleRoutine](https://rutinesandfit.atlassian.net/jira/software/projects/RF/boards/1?selectedIssue=RF-82)

## Type of change

- Feature (new feature of the server/application)

## Description

- creation of get, post and delete scheduleRoutines thunks
- added thunks on routine slice
- implementation of thunks on SelectRoutine and ScheduleRoutine screens
- routineDashboard screen integration

## Attachments

https://github.com/user-attachments/assets/9e907335-e91f-42b5-a8d0-693d1e1f5969

https://github.com/user-attachments/assets/31644685-cc83-4fb9-bd0b-69df19a51865

### CODE REVIEW

File: src/app/screens/do-routine/select-routine/index.tsx [hunk 1]
```diff
@@ -19,52 +20,29 @@ import { Props } from './types';
...
 
   const selectedRoutine = watch('routine');
 
   const filteredExercises = useMemo(() => {
-    return routineExercisesData.filter(
-      exercise => exercise.routine === selectedRoutine,
-    );
-  }, [routineExercisesData, selectedRoutine]);
+    return scheduleRoutines
+      .filter(routine => routine.day === selectedRoutine)
```
Comment (franco-marini) [inline src/app/screens/do-routine/select-routine/index.tsx]: Why are you doing this filter?

File: src/interfaces/routine.ts [hunk 2]
```diff
@@ -6,11 +6,14 @@ export interface Series {
...
+export interface RoutineExercise {
   id: string;
-  exerciseId: string;
-  name: string;
-  muscleGroup: string;
+  exercise: {
+    id: string;
+    name: string;
+    muscleGroup: string;
+    userId: string;
```
Comment (franco-marini) [inline src/interfaces/routine.ts]: This is doesn't required

**Review by franco-marini — COMMENTED** [review]
Nice work! Just 2 little comments to understand

File: src/app/screens/do-routine/select-routine/index.tsx (hunk 1 above)
Comment (AgustinCarthery) [reply src/app/screens/do-routine/select-routine/index.tsx]: because im displaying the exercises based on the current routine selected

Comment (AgustinCarthery) [reply src/app/screens/do-routine/select-routine/index.tsx:43]: Yes, im using due to the structure that we get on Routines: 
```
"exercises": [
                {
                    "id": "212428c6-fbe2-4c11-a800-93e5f84cf9ce",
                    "exerciseId": "1",
                    "name": "Remo con maquina",
                    "muscleGroup": "PECTORAL",
                    "repetitions": 5,
                    "restTimeSecs": 30,
                    "order": 1,
                    "series": [
                        {
                            "id": "39f2a6b9-c7ac-44fe-ade7-13e42f77a058",
                            "order": 1,
                            "weight": 80,
                            "weightMeasure": "kg",
                            "routineExerciseId": "212428c6-fbe2-4c11-a800-93e5f84cf9ce"
                        },
                ]
                
```
This is being used in the FlatList container: 
https://github.com/routine-fit/rosfit-mobile-app/blob/71982c2d294537a3e0d59da8647bddf7dca963a6/src/app/screens/do-routine/select-routine/styles.ts#L21

and in the renderExercise function: 
https://github.com/routine-fit/rosfit-mobile-app/blob/71982c2d294537a3e0d59da8647bddf7dca963a6/src/app/screens/do-routine/select-routine/index.tsx#L56

File: src/interfaces/routine.ts (hunk 2 above)
Comment (AgustinCarthery) [reply src/interfaces/routine.ts]: Yes, im using due to the structure that we get on payload: 

Screenshot 2024-10-09 at 14 56 49


This is being used in the FlatList container: 
https://github.com/routine-fit/rosfit-mobile-app/blob/71982c2d294537a3e0d59da8647bddf7dca963a6/src/app/screens/do-routine/select-routine/styles.ts#L21

and in the renderExercise function: 
https://github.com/routine-fit/rosfit-mobile-app/blob/71982c2d294537a3e0d59da8647bddf7dca963a6/src/app/screens/do-routine/select-routine/index.tsx#L56

Comment (franco-marini) [reply src/interfaces/routine.ts]: I mean the userId inside exercise is not required, sorry about that

**Review by franco-marini — APPROVED** [review]
🚀

**Pull request merged**